src/user-safe/UserSafeFactory.sol

```diff
@@ -17,13 +18,27 @@ contract UserSafeFactory is UpgradeableBeacon {
         address _owner
     ) UpgradeableBeacon(_implementation, _owner) {}
 
+    function getUserSafeAddress(bytes memory saltData, bytes memory data) external view returns (address) {
+        return CREATE3.predictDeterministicAddress(keccak256(abi.encode(saltData, data)));
+    }
+
     /**
      * @notice Function to deploy a new User Safe.
      * @param data Initialize function data to be passed while deploying a user safe.
      * @return Address of the user safe.
      */
-    function createUserSafe(bytes memory data) external returns (address) {
-        address safe = address(new BeaconProxy(address(this), data));
+    function createUserSafe(bytes memory saltData, bytes memory data) external returns (address) {
+        // address safe = address(new BeaconProxy(address(this), data));
+        address safe = address(
+            CREATE3.deployDeterministic(
+                abi.encodePacked(
+                    type(BeaconProxy).creationCode, 
+                    abi.encode(address(this), data)
+                ), 
+                keccak256(abi.encode(saltData, data))
```

Comment:
why do we need both `saldData` and `data`? not just `data`? 

https://github.com/Vectorized/solady/blob/main/src/utils/CREATE3.sol#L59

Reply:
Salt data will basically contain user's identification like UUID from the backend. Data is the init data for the contract that is being deployed. Do you want me to use do something like abi.encode(saltData, data) and send it as one param? Can do that but will increase some gas!

Reply:
all good :)